ant-node/src/networking/driver/cmd.rs

```diff
@@ -83,29 +82,42 @@ impl SwarmDriver {
                         trace!("Replicate cmd to self received, ignoring");
                     }
                 } else {
-                    if let Some(addrs) = addrs {
-                        // dial the peer and send the request
-                        if addrs.0.is_empty() {
-                            info!("No addresses for peer {peer:?} to send request. This could cause dial failure if swarm could not find the peer's addrs.");
-                        } else {
-                            let opts = DialOpts::peer_id(peer).addresses(addrs.0.clone()).build();
-
-                            match self.swarm.dial(opts) {
-                                Ok(()) => {
-                                    debug!("Dialing peer {peer:?} for req_resp with address: {addrs:?}",);
-                                }
-                                Err(err) => {
-                                    error!("Failed to dial peer {peer:?} for req_resp with address: {addrs:?} error: {err}",);
-                                }
+                    let addresses = if addrs.0.is_empty() {
+                        // The input addrs is a default one, try to fetch from local.
+                        let mut result: Vec<Multiaddr> = vec![];
+                        for kbucket in self.swarm.behaviour_mut().kademlia.kbuckets() {
+                            let peer_addrs = kbucket
+                                .iter()
+                                .filter_map(|peer_entry| {
+                                    if peer_entry.node.key.into_preimage() == peer {
+                                        Some(peer_entry.node.value.clone().into_vec())
+                                    } else {
+                                        None
+                                    }
+                                })
+                                .collect::<Vec<Vec<Multiaddr>>>();
+                            if !peer_addrs.is_empty() {
+                                result = peer_addrs[0].clone();
+                                break;
                             }
```

Comment:
Can we make this into its own separate helper function? It would help with readability A LOT.